x/smart/psiphon_dialer.go

```diff
@@ -10,24 +10,42 @@ import (
 	"fmt"
 	"os"
 	"path"
+	"runtime"
 
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 	"github.com/Jigsaw-Code/outline-sdk/x/psiphon"
 )
 
+func getUserCacheDir(finder *StrategyFinder, ctx context.Context) (string, error) {
+	var err error
+	var cacheBaseDir string
+	if runtime.GOOS == "android" {
+		cacheBaseDir, err = AndroidPrivateCacheDir()
+	} else {
+		// For every other system os.UserCacheDir works okay
+		cacheBaseDir, err = os.UserCacheDir()
+	}
+	if err != nil {
+		return "", fmt.Errorf("Failed to get the user cache directory: %w", err)
+	}
+
+	userCacheDir := path.Join(cacheBaseDir, "psiphon")
+	if err := os.MkdirAll(userCacheDir, 0700); err != nil {
+		return "", fmt.Errorf("Failed to create storage directory: %w", err)
+	}
+	finder.logCtx(ctx, "Using data store in %v\n", userCacheDir)
```

Comment:
Let's keep the cache file logic separate from the application logic. I would like us to reuse it in other places and it's cleaner/more maintainable.

That means we can remove the context parameter (which I think the style guide says it should be the first parameter).